Source/utils/file_util.cpp

```diff
@@ -73,6 +73,24 @@ bool FileExists(const char *path)
 #endif
 }
 
+bool IsWriteable(const char *path)
+{
+#if defined(_WIN64) || defined(_WIN32)
+	const auto pathUtf16 = ToWideChar(path);
+	if (pathUtf16 == nullptr) {
+		LogError("UTF-8 -> UTF-16 conversion error code {}", ::GetLastError());
+		return false;
+	}
+	uint32_t attr = ::GetFileAttributesW(&pathUtf16[0]);
+	if (attr != INVALID_FILE_ATTRIBUTES && (attr & FILE_ATTRIBUTE_READONLY) != 0) {
+		return false;
+	}
+	return true;

```

Comment:
Alternative, `<io.h>` on Windows has a `_waccess` function:

```suggestion
	return _waccess(&pathUtf16[0], 2) != -1);
```

https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/access-waccess?view=msvc-160

Not sure which is better